relayer/src/error.rs

```diff
@@ -187,6 +218,60 @@ pub enum Kind {
     },
 }
 
+impl RetryableError for Kind {
+    fn is_retryable(&self) -> bool {
+        // TODO: actually classify whether an error kind is retryable
+        true
+    }
+}
+
+fn error_with_source(err: Kind, source: impl Into<BoxError>) -> Error {
+    Context::new(err, Some(source.into())).into()
+}
+
+fn error(err: Kind) -> Error {
+    Context::new(err, None).into()
+}
+
+pub fn send_error<T>(err: SendError<T>) -> Error
```

Comment:
Can we remove the `_error` suffix from these methods and move them into the `Kind` impl? This way the suffix is redundant and we can import them qualified by the `Kind` type.

Reply:
Actually I want to avoid that, because moving them into `impl` block makes all of them implicitly imported. With explicit import, we can just look at the import list and know what errors can be raised in a particular function.

Ideally the error domain of a function should exactly match the enumerations in the error type, however that is not the case right now in our code base. For example, the function in `ChainRuntime` only raise a `SendError`, but the type signature indicates that it can send arbitrary errors. We should probably refactor that in the future, but for now the explicit import list serves as a good way for static inspection.